js/sagas/onboarding.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * A saga that manages the Onboarding.
+ *
+ * @flow
+ */
+
+import { type Saga } from 'redux-saga'
+import { takeLatest, select, take, call, put } from 'redux-saga/effects'
+import { NavigationActions } from 'react-navigation'
+
+import { type GlobalState } from '../reducers/types'
+import {
+  SESSION_INITIALIZE_SUCCESS,
+  TOS_ACCEPT_REQUEST,
+  TOS_ACCEPT_SUCCESS,
+  PIN_CREATED_REQUEST
+} from '../store/actions/constants'
+import ROUTES from '../navigation/routes'
+
+// The Pin step of the Onboarding
+function* pinStep(): Saga<void> {
```

Comment:
I would be more explicit on the intention here and call this `pinInputStep` - also is this step valid for the PIN update too? or is it just for the first input? from the comment it looks like it's limited to the onboarding process